kythe/typescript/indexer.ts

```diff
@@ -1471,6 +1500,38 @@ class Visitor {
     return vname;
   }
 
+  /**
+   * Emits a code fact for a variable declaration, specifying how the
+   * declaration should be presented to users.
+   *
+   * The form of the code fact is
+   *     ((local var)|const|let) <name>: <type>( = <initializer>)?
+   * where `(local var)` is the declaration of a variable in a catch clause.
+   */
+  emitVariableDeclarationCode(decl: ts.VariableDeclaration, declVName: VName) {
+    const codeParts: MarkedSource[] = [];
+    const initializerList = decl.parent;
+    const declKw = initializerList.kind === ts.SyntaxKind.CatchClause ?
+        '(local var)' :
+        initializerList.flags & ts.NodeFlags.Const ? 'const' : 'let';
+    const ty = this.typeChecker.getTypeAtLocation(decl);
+    const tyStr = this.typeChecker.typeToString(ty, decl);
+    codeParts.push(makeMarkedSource({kind: 'CONTEXT', preText: declKw}));
+    codeParts.push(makeMarkedSource({kind: 'BOX', preText: ' '}));
+    codeParts.push(
+        makeMarkedSource({kind: 'IDENTIFIER', preText: decl.name.getText()}));
+    codeParts.push(
+        makeMarkedSource({kind: 'TYPE', preText: ': ', postText: tyStr}));
```

Comment:
We put this in the same node so the codesearch UI formats the type as `: ty`. With the identifier this becomes `id: ty`; otherwise it would be `idty`.